tikv-client/src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -349,6 +349,24 @@ public ExecutorService getThreadPoolForDeleteRange() {
     return res;
   }
 
+  public ExecutorService getThreadPoolForIsAlive() {
+    if (isAliveThreadPool == null) {
+      synchronized (this) {
+        if (isAliveThreadPool == null) {
+          isAliveThreadPool =
+              new ThreadPoolExecutor(
+                  20,
+                  20,
+                  0,
+                  TimeUnit.MILLISECONDS,
+                  new ArrayBlockingQueue<>(1),
+                  new ThreadPoolExecutor.DiscardPolicy());
+        }
+      }
+    }
+    return isAliveThreadPool;
+  }
```

Comment:
Some questions:
1. one pool for one session?
2. is 20 threads necessary?
3. the blocking queue has only 1 slot, is it right?

Reply:
1. Yes. one pool for one session, just like other pool in Tisession `indexScanThreadPool`,tableScanThreadPool etc.
2. There may be too many probe tasks. If there is only one thread, the latest task may be discarded.
3. The task execution speed may be slower than the task generated speed. I use one slot so that the poll can discard the tasks. If we use too many slots, The task may be outdated. We just need the latest task.
4. The same as 3. Newfixedthreadpool's blocking queue has too many slots.

all in all. it may be difficult to find the best parameters for the poll. If this change can not meet our need. I will consider using a corn job. (may use newScheduledThreadPool  in java)